CMakeLists.txt

```diff
@@ -1,26 +1,82 @@
-cmake_minimum_required (VERSION 3.14)
-project (ifc-sdk)
+# Copyright Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-# Fetch Microsoft GSL
-include(FetchContent)
+cmake_minimum_required(VERSION 3.14)
```

Comment:
There is no feature that requires that version. `TODO.md` mentions the project using modules itself, but that would require 3.28 for portable support.

Reply:
3.28 isn't out yet, so I would take 3.27 :-)

That would allow people to experiment on the side with modularization with the compilers for which the support is already there.  This isn't a required change for this PR, but since you were in the neighborhood... :-)

Reply:
Not a fan of unnecessarily raising the minimum required version. Using the experimental modules support requires file sets to be used and normal variables to be set, so the contributor will have to edit the project files regardless.

Reply:
> Not a fan of unnecessarily raising the minimum required version.

Understood; that is part of the experimental nature of the project :-)

> Using the experimental modules support requires file sets to be used and normal variables to be set, so the contributor will have to edit the project files regardless.

That is true, the CML file would need to be edited anyway but knowing we are all working from the same minimum required version is always good to know when sharing results of experiments.  For instance, CMake has gotten strict in some areas while other features have been deprecated, new warnings introduced, etc.  Having the same baseline helps.